libmamba/src/core/repo.cpp

```diff
@@ -101,11 +94,23 @@ namespace mamba
 
     void MRepo::init(MPool& pool)
     {
-        m_repo->appdata = this;
         m_real_repo_key = pool_str2id(pool, "solvable:real_repo_url", 1);
+        m_mrepo_key = pool_str2id(pool, "solvable:mrepo_url", 1);
```

Comment:
What is the difference between `repo` and `real_repo` ?

Reply:
I'd like to know too! This is a mechanical operation that stores the value of `MRepo::url` for retrival.
I tried to merge `real_repo_url` and `mrepo_url` but it was significantly more tricky than the scope of this PR.

Reply:
There is also `m_metadata.url`/"mamba:url" in the mix